File: ios/MazadatVideoOverlay.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(MazadatVideoOverlay, NSObject)

RCT_EXTERN_METHOD(multiply:(float)a b:(float)b
                 resolve:(RCTPromiseResolveBlock)resolve
                 reject:(RCTPromiseRejectBlock)reject)

RCT_EXTERN_METHOD(playVideo:(NSString *)link
                 resolve:(RCTPromiseResolveBlock)resolve
                 reject:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
  return NO;
}

@end

File: ios/MazadatVideoOverlay.swift

import Foundation
import UIKit
@objc(MazadatVideoOverlay)
class MazadatVideoOverlay: NSObject {

  @objc
  func multiply(_ a: Float, b: Float, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) {
      
    resolve(a+b)
  }
    
    @objc
    func playVideo(_ link: String, resolve:RCTPromiseResolveBlock,reject:RCTPromiseRejectBlock) {
        DispatchQueue.main.async{
            let controller=VideoPlayerController()
            controller.setLink(link: link)
            controller.modalPresentationStyle = .overCurrentContext
            UIApplication.shared.windows.filter {$0.isKeyWindow}.first?.rootViewController?.present(controller, animated: true, completion: nil)
        }
      
    }
}
